src/SeleniumLibrary/keywords/waiting.py

```diff
@@ -135,15 +134,26 @@ def wait_until_page_contains_element(self, locator, timeout=None,
 
         ``error`` can be used to override the default error message.
         """
-        self._wait_until(
-            lambda: self.find_element(locator, required=False) is not None,
-            "Element '%s' did not appear in <TIMEOUT>." % locator,
-            timeout, error
-        )
+        if is_noney(limit):
+            self._wait_until(
+                lambda: self.find_element(locator, required=False) is not None,
+                "Element '%s' did not appear in <TIMEOUT>." % locator,
+                timeout, error
+            )
+        else:
+            limit = int(limit)
+            self._wait_until(
+                lambda: len(self.find_elements(locator)) is limit,
+                'Page should have contained "{}" element(s),'
```

Comment:
Would the error message be more readable if the the RF [plural_or_not](https://github.com/robotframework/robotframework/blob/master/src/robot/utils/misc.py#L106) would be used to format ´s` in the end correctly?